src/pointer-analysis/value_set.cpp

```diff
@@ -367,6 +367,45 @@ static std::string strip_first_field_from_suffix(
   return suffix.substr(field.length() + 1);
 }
 
+const value_sett::entryt *value_sett::get_entry_for_symbol(
+  const irep_idt identifier,
+  const typet &expr_type,
+  const std::string &suffix) const
+{
+  // We only deal with pointers, integers, arrays and structs
+  if(
+    expr_type.id() != ID_pointer && expr_type.id() != ID_signedbv &&
+    expr_type.id() != ID_unsignedbv && expr_type.id() != ID_struct &&
+    expr_type.id() != ID_union && expr_type.id() != ID_array)
```

Comment:
Now that this is factored out you will need to consider `ID_struct_tag` and `ID_union_tag` as well, and then use `ns.follow_tag`.